src/features/profile/pages/ValidateEmailChange/ValidateEmailChange.native.test.tsx

```diff
@@ -54,29 +67,53 @@ describe('ValidateEmailChange', () => {
     expect(screen.getByText('john@doe.com')).toBeTruthy()
   })
 
-  it('should redirect to TrackEmailChange if submit is success', async () => {
+  it('should sign out if submit is success and user is logged in', async () => {
     render(<ValidateEmailChange navigation={navigation} route={route} />)
 
     fireEvent.press(screen.getByText('Valider l’adresse e-mail'))
 
     await waitFor(() => {
-      expect(navigation.navigate).toHaveBeenNthCalledWith(1, 'TrackEmailChange')
+      expect(mockSignOut).toHaveBeenCalledTimes(1)
     })
   })
 
-  it('should display a snackbar if submit triggers an error', async () => {
-    emailUpdateValidateSpy.mockRejectedValueOnce('Oops test')
+  it('should not sign out if submit is success and user is logged in', async () => {
+    mockUseAuthContext.mockReturnValueOnce({
+      isLoggedIn: false,
+      setIsLoggedIn: jest.fn(),
+      isUserLoading: false,
+      refetchUser: jest.fn(),
+    })
+    render(<ValidateEmailChange navigation={navigation} route={route} />)
+
+    fireEvent.press(screen.getByText('Valider l’adresse e-mail'))
+
+    await waitFor(() => {
+      expect(mockSignOut).not.toHaveBeenCalled()
+    })
```

Comment:
Obligé de mettre dans un `waitFor` le fait que ça n'arrivera pas ?
```suggestion
    expect(mockSignOut).not.toHaveBeenCalled()
```

Reply:
Yes on se prend un warning, la validation reste asynchrone